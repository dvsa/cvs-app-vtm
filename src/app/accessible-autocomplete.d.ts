declare module 'accessible-autocomplete/dist/accessible-autocomplete.min' {
  interface ElementParam {
    element: HTMLElement;
  }
  interface EnhancedElementParam {
    selectElement: HTMLElement | null;
  }
  interface DefaultParams {
    id?: string;
    source?: Array<any>;
    onConfirm?: (value: string) => void;
    confirmOnBlur?: boolean;
    required?: true;
    defaultValue?: string;
    autoselect?: boolean;
    showAllValues?: boolean;
    dropdownArrow?: Function;
  }

  export interface AutocompleteParams extends DefaultParams, ElementParam {}
  export interface AutocompleteEnhanceParams extends DefaultParams, EnhancedElementParam {}

  export default function accesibleAutocomplete(params: AutocompleteParams): void;
  export function enhanceSelectElement(params: AutocompleteEnhanceParams): void;
}
