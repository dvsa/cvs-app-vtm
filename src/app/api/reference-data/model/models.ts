export * from './reference-data.model';
