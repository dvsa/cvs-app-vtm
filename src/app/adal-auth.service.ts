import { Injectable } from '@angular/core';
import { MsAdalAngular6Service } from 'microsoft-adal-angular6';
// import { MsAdalAngular6Module } from 'microsoft-adal-angular6';


@Injectable({
  providedIn: 'root'
})
export class AdalAuthService {

  constructor(
    private msAdal: MsAdalAngular6Service
  ) { }

  isAuthenticated() {
    return this.msAdal.isAuthenticated
  }

  getProfile() {
    return this.msAdal.userInfo.profile
  }

  getResourceForEndpoint(url) {
    return this.msAdal.GetResourceForEndpoint(url)
  }

  acquireToken(resource) {
    return this.msAdal.acquireToken(resource)
  }

  accessToken() {
    return this.msAdal.accessToken
  }
}
