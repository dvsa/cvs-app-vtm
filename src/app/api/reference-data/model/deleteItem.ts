/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

export interface DeleteItem { 
    success?: boolean;
}