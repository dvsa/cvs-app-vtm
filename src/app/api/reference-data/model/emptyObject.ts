/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyObject {
}
