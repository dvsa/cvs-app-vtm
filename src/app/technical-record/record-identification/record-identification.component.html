<ng-container *ngIf="!editState">
  <dl class="govuk-summary-list">
    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">
        Vehicle identification number (VIN)
      </dt>
      <dd id="test-vin" class="govuk-summary-list__value">
        {{ vehicleTechRecord.vin }}
      </dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">
        Vehicle registration mark (VRM)
      </dt>
      <dd id="test-vrm" class="govuk-summary-list__value">
        <div *ngIf="vehicleTechRecord.primaryVrm; else hyphen">
          <p>{{ vehicleTechRecord.primaryVrm }}</p>
        </div>
      </dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">
        Previous VRM
      </dt>
      <dd id="test-secondaryVrms" class="govuk-summary-list__value">
        <div *ngIf="vehicleTechRecord.secondaryVrms.length; else hyphen">
          <p *ngFor="let vrmSecondary of vehicleTechRecord.secondaryVrms; let i = index">
            <span id="{{ 'test-vrm-' + i }}">{{ vrmSecondary }}</span>
          </p>
        </div>
      </dd>
    </div>

    <div class="govuk-summary-list__row" *ngIf="isTrailer">
      <dt class="govuk-summary-list__key main-details">
        Trailer ID
      </dt>
      <dd id="test-trailerId" class="govuk-summary-list__value">
        {{ vehicleTechRecord.trailerId }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>
  </dl>
</ng-container>

<ng-template #hyphen><span>-</span></ng-template>

<!-- Edit Vehicle Identification -->
<ng-container *ngIf="editState">
  <vtm-record-identification-edit
    [vin]="vehicleTechRecord.vin"
    [primaryVrm]="vehicleTechRecord.primaryVrm"
  ></vtm-record-identification-edit>
</ng-container>
