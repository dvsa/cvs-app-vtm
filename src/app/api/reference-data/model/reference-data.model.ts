import { ReferenceDataResourceType } from '@models/reference-data.model';

export type ReferenceDataItem = {
  resourceType: ReferenceDataResourceType;
  resourceKey: string | number;
};

export type ReferenceDataList = Array<ReferenceDataItem>;

export type ReferenceDataApiResponse = {
  data: ReferenceDataList;
};
