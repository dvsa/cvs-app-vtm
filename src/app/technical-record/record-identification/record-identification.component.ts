import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';

import { VehicleTechRecordEdit } from '@app/models/vehicle-tech-record.model';

@Component({
  selector: 'vtm-record-identification',
  templateUrl: './record-identification.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class RecordIdentificationComponent implements OnInit {
  @Input() vehicleTechRecord: VehicleTechRecordEdit;
  @Input() editState: boolean;

  isTrailer: boolean | string;

  constructor() {}

  ngOnInit() {
    this.isTrailer = this.vehicleTechRecord.trailerId || false;
  }
}
