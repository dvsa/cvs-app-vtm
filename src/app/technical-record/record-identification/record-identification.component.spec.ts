import { CUSTOM_ELEMENTS_SCHEMA, Component, Input } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { VIEW_STATE } from '@app/app.enums';
import { RecordIdentificationComponent } from './record-identification.component';
import { VehicleTechRecordEdit } from '@app/models/vehicle-tech-record.model';

describe('RecordIdentificationComponent', () => {
  let component: RecordIdentificationComponent;
  let fixture: ComponentFixture<RecordIdentificationComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [RecordIdentificationComponent, TestRecordIdentificationEditComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(RecordIdentificationComponent);
    component = fixture.componentInstance;
    component.vehicleTechRecord = {
      vin: 'ABCDEFGH654321',
      primaryVrm: 'LKJH654',
      secondaryVrms: ['AAJH654', 'LKJHBBB']
    } as VehicleTechRecordEdit;
  });

  it('should create view only with populated data', () => {
    component.editState = false;
    fixture.detectChanges();

    expect(component).toBeDefined();
    expect(fixture).toMatchSnapshot();
  });

  it('should create view only with trailer Id data', () => {
    component.vehicleTechRecord.trailerId = '0285678';
    component.editState = false;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should render the editable component if editState is true', () => {
    component.editState = true;
    component.viewState = VIEW_STATE.CREATE;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});

@Component({
  selector: 'vtm-record-identification-edit',
  template: `
    <div>Vin is {{ vin }}</div>
    <div>Primary Vrm is {{ primaryVrm }}</div>
    <div>View State is {{ viewState }} Create state (2)</div>
  `
})
class TestRecordIdentificationEditComponent {
  @Input() vin: string;
  @Input() primaryVrm: string;
  @Input() viewState: VIEW_STATE;
}
