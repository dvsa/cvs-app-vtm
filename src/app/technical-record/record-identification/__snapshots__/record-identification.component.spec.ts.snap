// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordIdentificationComponent should create view only with populated data 1`] = `
<vtm-record-identification
  editState="false"
  vehicleTechRecord={[Function Object]}
>
  <dl
    class="govuk-summary-list"
  >
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Vehicle identification number (VIN) 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-vin"
      >
         ABCDEFGH654321 
      </dd>
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Vehicle registration mark (VRM) 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-vrm"
      >
        
        <div>
          <p>
            LKJH654
          </p>
        </div>
      </dd>
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Previous VRM 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-secondaryVrms"
      >
        
        <div>
          
          <p>
            <span
              id="test-vrm-0"
            >
              AAJH654
            </span>
          </p>
          <p>
            <span
              id="test-vrm-1"
            >
              LKJHBBB
            </span>
          </p>
        </div>
      </dd>
    </div>
  </dl>
</vtm-record-identification>
`;

exports[`RecordIdentificationComponent should render the editable component if editState is true 1`] = `
<vtm-record-identification
  editState={[Function Boolean]}
  vehicleTechRecord={[Function Object]}
>
  <vtm-record-identification-edit
    ng-reflect-primary-vrm="LKJH654"
    ng-reflect-vin="ABCDEFGH654321"
  >
    <div>
      Vin is ABCDEFGH654321
    </div>
    <div>
      Primary Vrm is LKJH654
    </div>
  </vtm-record-identification-edit>
</vtm-record-identification>
`;
